migration_guide_iothub.md

```diff
@@ -1,237 +1,222 @@
+**Note that currently V3 only presents an async set of APIs. This guide will be updated when that changes**
 
-For changes to the `ProvisioningDeviceClient` please refer to `migration_guide_provisioning.md` in this same directory.
+For changes when using the Device Provisioning Service, please refer to `migration_guide_provisioning.md` in this same directory.
 
-## Connecting to IoTHub
-One of the primary changes in V3 is the removal of automatic connections when invoking other APIs on the `IoTHubDeviceClient` and `IoTHubModuleClient`. You must now make an explicit manual connection before sending or receiving any data.
+The design goals for V3 were to make a more stripped back, simple API surface that allows for a greater flexibility for the end user, as well as improved reliability and clarity. We have attempted to remove as much implicit behavior as possible in order to give full control of functionality to the end user. Additionally, we have attempted to make the experience of using the API simpler to address common pitfalls, and make applications easier to write.
 
-### V2
+## Connection Management
+The most significant change in V3 is the removal of manual connection/disconnection. Connections are now managed automatically by a context manager.
+
+#### V2
 ```python
-from azure.iot.device import IoTHubDeviceClient
+from azure.iot.device.aio import IoTHubDeviceClient
 
-client = IoTHubDeviceClient.create_from_connection_string("<Your Connection String>")
-client.send_message("some message")
+async def main():
+    client = IoTHubDeviceClient.create_from_connection_string("<Your Connection String>")
+    await client.connect()
+    # <do things>
+    await client.disconnect()
 ```
 
-### V3
+#### V3
 ```python
-from azure.iot.device import IoTHubDeviceClient
+from azure.iot.device import IoTHubSession
 
-client = IoTHubDeviceClient.create_from_connection_string("<Your Connection String>")
-client.connect()
-client.send_message("some message")
+async def main():
+    async with IoTHubSession.from_connection_string("<Your Connection String>") as session:
+        # <do things>
 ```
 
-Note that many people using V2 may already have been doing manual connects, as for some time, this has been our recommended practice.
-
-Note also that this change does *not* affect automatic reconnection attempts in the case of network failure. Once the manual connect has been successful, the client will (under default settings) still attempt to retain that connected state as it did in V2.
-
+When the context manager is entered, a connection will be established before running the block of code inside the context manager. After the block is done executing, a disconnection will occur upon context manager exit. You can consider that all code within the block is written with the expectation of a connection, as the context manager
+represents a connection to the IoT Hub.
```

Comment:
Suggestion: Since markdown concatenates consecutive lines, put each sentence on it's own line. It renders the same, but it's easier to review and easier to track changes.